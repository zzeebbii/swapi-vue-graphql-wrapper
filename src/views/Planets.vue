<template>
  <div>
    <div v-if="loading" class="loader"></div>
    <Search @onSearch="onSearch" />
    <div class="error" v-if="error">{{ error }}</div>
    <div class="cards">
      <Card
        v-for="planet in planets"
        :url="'planet/' + planet.id"
        :title="planet.name"
        :info="`Diameter: ${planet.diameter} | Climate: ${planet.climate}`"
        :key="planet.name"
      />
    </div>
  </div>
</template>

<script>
// @ is an alias to /src
import Card from "@/components/Card.vue";
import Search from "@/components/Search.vue";
import axios from "axios";
import { ALL_PLANETS_QUERY, SEARCH_PLANET_QUERY } from "@/utils/constants";

export default {
  name: "Planets",
  data: function() {
    return {
      planets: [],
      loading: 0,
      planetToSearch: "",
      skipQuery: true,
      error: null
    };
  },
  components: {
    Card,
    Search
  },
  apollo: {
    planets: {
      query: ALL_PLANETS_QUERY,
      error(error) {
        this.error = JSON.stringify(error.message);
      }
    },
    searchPlanet: {
      query: SEARCH_PLANET_QUERY,
      variables() {
        return {
          name: this.planetToSearch
        };
      },
      skip() {
        return this.skipQuery;
      },
      manual: true,
      result({ data, loading }) {
        if (!loading) {
          this.planets = data.planetByName;
        }
      },
      error(error) {
        this.error = JSON.stringify(error.message);
      }
    }
  },
  methods: {
    onSearch(value) {
      this.planetToSearch = value;
      this.skipQuery = false;
      this.$apollo.queries.searchPlanet.refetch();
    }
  }
};
</script>
