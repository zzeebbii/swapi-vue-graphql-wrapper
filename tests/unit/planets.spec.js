import { shallowMount } from "@vue/test-utils";
import Planets from "@/views/Planets.vue";


describe("Planets.vue", () => {
  it("mounted should set proper data from AJAX response [success]", function() {
    const wrapper = shallowMount(Planets, {
      data() {
        return {
          planets: [
            {
              id: 1,
              name: "Test 1",
              diameter: 100,
              climate: "warm"
            },
            {
              id: 2,
              name: "Test 2",
              diameter: 150,
              climate: "cold"
            }
          ]
        };
      }
    });
    expect(wrapper).toMatchSnapshot();
  });
});
