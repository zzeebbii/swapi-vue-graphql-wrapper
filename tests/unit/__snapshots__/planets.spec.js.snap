// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets.vue mounted should set proper data from AJAX response [success] 1`] = `
<div>
  <!---->
  <search-stub></search-stub>
  <!---->
  <div class="cards">
    <card-stub title="Test 1" info="Diameter: 100 | Climate: warm" url="planet/1"></card-stub>
    <card-stub title="Test 2" info="Diameter: 150 | Climate: cold" url="planet/2"></card-stub>
  </div>
</div>
`;
